### DISCUSSION TITLE
Docker image for psalm

### DISCUSSION THREAD

**Discussion by Legion112** [vimeo/psalm#6602] [Ideas]
I suggest adding the creation of a docker image for the psalm.
It will decouple a project from psalm dependency and allow to use of the latest version of the psalm even when there is a conflict with composer dependency. 
I can automate this process but need feedback first. Will it be beneficial to others? And how do the authors think about it?

**Comment by orklah** [comment]
Can you tell us more? How exactly would that change things for the end user?

**Reply by Legion112** [reply]
You would not need to install the psalm and php in order to run plasm in a project.
I know we can just downloaded the psalm.phar but it still will require to install the php and physically store the binary somewhere.
Image with psalm will minimize the afford to add psalm to any CI pipeline.

**Reply by orklah** [reply]
What I meant is, would that be a new repo on packagist or maybe even just an docker image in a hub somewhere? 

Personnally, I have no problem having Psalm available as a docker image if it's something aside we provide as extra, but I wouldn't want that to be something that replace the current way of doing things.

**Reply by Legion112** [reply]
No I just offering to setup build psalm imagine on new tag. And publishing it to dockerhub. I can make a PR with GitHubAction.
But it will also need to setup the secret for publishing images on dockerhub... I would need the help from the mainteiner on that side.

**Comment by Legion112** [comment]
The command will look something like that
`docker run -it --rm -v $(pwd):/app:ro --workdir=/app fearofcode/commision-psalm:0.0 psalm --no-cache`
There are not need for additional command to download psalm. Just on command will do the download and execution.

**Reply by orklah** [reply]
@weirdan what do you think? Do you have any experience with docker?

If we want do do that, we would probably prefer to distribute it under Psalm vendor name to show it's official (or under Vimeo's name? I don't know the legal side of things here...)

**Comment by weirdan** [comment]
It appears we already ship a docker image at https://hub.docker.com/r/vimeo/psalm-github-actions (repo: https://github.com/psalm/psalm-github-actions). Can't we reuse that repo instead?

**Reply by Legion112** [reply]
That nice. Then I suggest to add the description in readme file with simple instructions. Or in any documentation because I was not able to find that repo by "docker"  key word

**Reply by orklah** [reply]
By all means, please submit a PR, it's always good to have more documentation

**Reply by Legion112** [reply]
Sure

**Comment by paulshryock** [comment]
I would love to have an official Psalm Docker image (with latest or various versions of PHP, Composer, Psalm) that I could use in a BitBucket Pipeline. Or would love to have an official BitBucket Pipeline similar to the GitHub Action above.

**Reply by Legion112** [reply]
You could use the github action image for bitbucket as well

**Reply by paulshryock** [reply]
> You could use the github action image for bitbucket as well

Looks like that repo uses [php 7.4](https://github.com/psalm/psalm-github-actions/blob/da14b2e266d252c8455fa9eaedf1476234c0b146/Dockerfile#L1); my repo requires php 8.1

**Reply by Legion112** [reply]
Yeah seem nobody supports it.
@weirdan I think it's reasonable to create psalm image build process here. 
I could create such merge request but wanna have support on that idea.

**Reply by weirdan** [reply]
No objections from me, as long as you're happy to provide reasonable support (monitoring issues and providing fixes when you have time) for it going forward.

**Comment by Legion112** [comment]
Created pull request 
https://github.com/vimeo/psalm/pull/7700/files
I was not able to build phar successfully. (After running psalm image with phar) was getting errors.
Refactoring to phar could decrease the size of the image, for now, is not that important.

**Reply by Legion112** [reply]
Moved to another PR.
https://github.com/vimeo/psalm/pull/7701
